doc/reference/dev/debug.md

```diff
@@ -1,5 +1,130 @@
+
+Occasionally, if you are using a `(not safe)` declaration in some
+performance critical piece of code, there can be segfaults. All the
+standard Gerbil modules included in `libgerbil` have debug symbols
+enabled, so you can just run the program in gdb and get the location
+of the crash.
+
+If the crash is inside `libgerbil`, then you will most like have exact
+location pointing to the _expanded_ source of the crash. You can find
+the offending code in `GERBIL_PREFIX/lib/static`.  Note that we
+_could_ have the tracked scheme source point to the original,
+unexpanded source, but we prefer to work with expanded source as we'd
+like to see the actual code that was compiled and also have the
+ability to debug potential compiler bugs (these are exceedingly rare,
+but they can happen).
+
+If the crash is inside your own library or main code and you have an
+input repro, you can just load the executable module of your program
+dynamically with `gerbil :your-executable-module input-param
+...`.
+So within gdb, you can `gdb gerbil` and `r :your-executable-module input-param`.
+This will get you source location, as all dynamically executable
+modules are compiled with debug symbols.
+
+If the crash is non deterministic, you can build your executable with
+`gerbil build --debug`. This will get you debug symbols for the binary
+and you can just run it directly with gdb until the crash occurs.  The
+actual expanded source files will be in your build's
+`GERBIL_PATH/lib/static`.  So if you built your project with `gerbil
+build --debug`, these will be located in `.gerbil/lib/static`.
+
+::: warning Note
+Sometimes you might get a crash that has no usable stack trace.  This
+can occur if you are calling a non procedure in unsafe context or
+because of some very low level bugs.  If that happens, you will need
+to guess a bit where the crash is coming from and disable your `(not
+safe)` declarations in the relevant module so that you get an
+exception with a stack trace instead.
+
+If you determine that such a crash occurs because of some low level
+Gerbil or Gambit bug, please file an [Issue](https://github.com/mighty-gerbils/gerbil/issues)
+and/or get in touch with the Gerbil development team on
+[gitter](https://gitter.im/gerbil-scheme/community).
+:::
+
+
+## Debugging Live Programs
+
+You can use a remote/network REPL to connect to a running program
+and evaluate code that helps you inspect and modify the state of
+the program.
+
+Gerbil offers a couple of different options for this:
+ - The embedded [Network REPL](/reference/std/net/repl.md) is useful for any
+   running program in your development workstation or laptop.
+   You can just use `telnet` to connect, but you have to
+   manually start the repl server inside your program.
+ - Servers in an [Actor Ensemble](/tutorials/ensemble.md) automatically have
+   network repl capabilities for debugging and loading code.
+   You can interact with them  using the `gxensemble` tool.
```

Comment:
Remove extra spaces between `them  using` ? There may be more extra spaces everywhere to regexp away.

Reply:
sure.